[receiver/mysql] Flaky integration test

### Component(s)

receiver/mysql

### What happened?

https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4089471343/jobs/7052232713#step:6:544

```
    integration_test.go:70: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/mysqlreceiver/integration_test.go:70
        	Error:      	Received unexpected error:
        	            	resource "map[mysql.instance.endpoint:localhost:3306]": scope "otelcol/mysqlreceiver": number of metrics doesn't match expected: 21, actual: 20
        	Test:       	TestMySqlIntegration/Running_mysql_version_8.0
--- FAIL: TestMySqlIntegration (0.00s)
    --- FAIL: TestMySqlIntegration/Running_mysql_version_8.0 (31.22s)
FAIL
```

origin from https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/18284

### Collector version

N/A

### Environment information

## Environment
OS: (e.g., "Ubuntu 20.04")
Compiler(if manually compiled): (e.g., "go 14.2")


### OpenTelemetry Collector configuration

_No response_

### Log output

_No response_

### Additional context

_No response_

--- a/receiver/mysqlreceiver/integration_test.go
+++ b/receiver/mysqlreceiver/integration_test.go
@@ -47,6 +47,7 @@ func TestMySqlIntegration(t *testing.T) {
 
 		f := NewFactory()
 		cfg := f.CreateDefaultConfig().(*Config)
+		cfg.CollectionInterval = time.Second
 		cfg.Endpoint = net.JoinHostPort(hostname, "3306")
 		cfg.Username = "otel"
 		cfg.Password = "otel"
@@ -61,15 +62,21 @@ func TestMySqlIntegration(t *testing.T) {
 		}, 2*time.Minute, 1*time.Second, "failed to receive more than 0 metrics")
 		require.NoError(t, rcvr.Shutdown(context.Background()))
 
-		actualMetrics := consumer.AllMetrics()[0]
-
 		expectedFile := filepath.Join("testdata", "integration", "expected.8_0.yaml")
 		expectedMetrics, err := golden.ReadMetrics(expectedFile)
 		require.NoError(t, err)
 
-		require.NoError(t, pmetrictest.CompareMetrics(expectedMetrics, actualMetrics,
-			pmetrictest.IgnoreMetricValues(), pmetrictest.IgnoreMetricDataPointsOrder(),
-			pmetrictest.IgnoreStartTimestamp(), pmetrictest.IgnoreTimestamp()))
+		compareOpts := []pmetrictest.CompareMetricsOption{
+			pmetrictest.IgnoreMetricValues(),
+			pmetrictest.IgnoreMetricDataPointsOrder(),
+			pmetrictest.IgnoreStartTimestamp(),
+			pmetrictest.IgnoreTimestamp()}
+
+		require.Eventually(t, func() bool {
+			allMetrics := consumer.AllMetrics()
+			latestMetrics := allMetrics[len(allMetrics)-1]
+			return nil == pmetrictest.CompareMetrics(expectedMetrics, latestMetrics, compareOpts...)
+		}, 30*time.Second, time.Second)
 	})
 }
 

